### DISCUSSION TITLE
No libsmali.jsonl entry for /com/shaded/fasterxml/jackson, this library will be ignored.

### DISCUSSION THREAD

**Discussion by sudo-joe** [iBotPeaches/Apktool#2958] [Q&A]
Hello

tried to install Apktool 2.7.0 on LMint 20.3 but get the following error message:

mint@mint:/media/mint/BU/APK-Library-Scanner$ scanapk 446096481062076416.apk 
No libsmali.jsonl entry for /com/shaded/fasterxml/jackson, this library will be ignored.

No offending libs found.

Question:
->No libsmali.jsonl entry for /com/shaded/fasterxml/jackson, this library will be ignored.
What am I doing wrong? How can this error message be prevented?

->No offending libs found.
This is very hardly possible because https://reports.exodus-privacy.eu.org/de/reports/291329/
did found 8 Trackers......

Thank you very much for any help!
Joe

-------------------------------------Installation of APKtool 2.7.0------------------------------------
sudo apt-get update

sudo apt install php-cli php-json

git clone https://gitlab.com/IzzyOnDroid/repo.git

mkdir -p repo/lib/radar/tool
cd repo/lib/radar/tool

wget -q https://github.com/iBotPeaches/Apktool/releases/download/v2.7.0/apktool_2.7.0.jar

wget -q https://github.com/iBotPeaches/Apktool/raw/master/scripts/linux/apktool

ln -s apktool_2.7.0.jar apktool.jar
chmod u+x apktool

cd ../../../bin
alias scanapk="$(pwd)/scanapk.php"

mint@mint:/media/mint/BU/APK-Library-Scanner$ scanapk 446096481062076416.apk  
No libsmali.jsonl entry for /com/shaded/fasterxml/jackson, this library will be ignored.

No offending libs found.

**Comment by iBotPeaches** [comment]
What happens when you just execute `java -jar apktool.jar` ? Not really following what all this wrapper stuff is.

**Reply by sudo-joe** [reply]
mint@mint:/media/mint/BU/APK-Library-Scanner$ sudo java -jar apktool_2.7.0.jar
Apktool v2.7.0 - a tool for reengineering Android apk files
with smali v2.5.2-403e9037 and baksmali v2.5.2-403e9037
Copyright 2010 Ryszard Wiśniewski <brut.alll@gmail.com>
Copyright 2010 Connor Tumbleson <connor.tumbleson@gmail.com>

usage: apktool
 -advance,--advanced   prints advance information.
 -version,--version    prints the version then exits
usage: apktool if|install-framework [options] <framework.apk>
 -p,--frame-path <dir>   Stores framework files into <dir>.
 -t,--tag <tag>          Tag frameworks using <tag>.
usage: apktool d[ecode] [options] <file_apk>
 -f,--force              Force delete destination directory.
 -o,--output <dir>       The name of folder that gets written. Default is apk.out
 -p,--frame-path <dir>   Uses framework files located in <dir>.
 -r,--no-res             Do not decode resources.
 -s,--no-src             Do not decode sources.
 -t,--frame-tag <tag>    Uses framework files tagged by <tag>.
usage: apktool b[uild] [options] <app_path>
 -f,--force-all          Skip changes detection and build all files.
 -o,--output <dir>       The name of apk that gets written. Default is dist/name.apk
 -p,--frame-path <dir>   Uses framework files located in <dir>.

For additional info, see: https://ibotpeaches.github.io/Apktool/ 
For smali/baksmali info, see: https://github.com/JesusFreke/smali
mint@mint:/media/mint/BU/APK-Library-Scanner$

**Comment by IzzySoft** [comment]
This is no problem caused by Apktool. That message comes from my library scanner (`scanapk`) which @sudo-joe is using there. And it says exactly that: there's no details for `/com/shaded/fasterxml/jackson` in `libsmali.jsonl`. While that library was encountered in several APKs and has a record in `libinfo.jsonl` (you can look it up there), the corresponding entry pointing to the source (where can you find it) is missing as I wasn't able to find those yet.

```json
{"id": "/com/shaded/fasterxml/jackson", "emphasize": 0, "details": "modules to process data encoded in Avro, BSON, CBOR, CSV, Smile, (Java) Properties, Protobuf, XML or YAML.", "mwid": "", "anti": []}
```

> What am I doing wrong?

Nothing (apart from reporting it at the wrong place maybe; `scanapk` is part of [this repo](https://gitlab.com/IzzyOnDroid/repo/) :wink:).

> How can this error message be prevented?

If someone can tell me where the sources of that library are, I can add it to `libsmali.jsonl`. Alternatively, if that message annoys you, you can simply remove the quoted line from `libinfo.jsonl` – or comment out [this line](https://gitlab.com/IzzyOnDroid/repo/-/blob/master/bin/scanapk.php?ref_type=heads#L44) in `scanapk.php`.

@iBotPeaches I'm not an Android dev, so I'm not sure if my assumption here is correct: does the `/shaded/` part in the name point to a relocation using the Apache Maven Shade Plugin, and it would be safe to assign that entry to `/com/fasterxml/jackson`? And would that generally apply to this pattern (i.e. it's safe to assign the `/shaded/` entries to their "parents")? Then I'd fix that accordingly with the next update.
